py-polars/tests/unit/datatypes/test_temporal.py

```diff
@@ -760,6 +765,112 @@ def test_upsample_time_zones(
     assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    ("every", "offset", "fill", "expected"),
+    [
+        (
+            "1i",
+            "0i",
+            "forward",
+            pl.DataFrame(
+                {
+                    "index": [1, 2, 3, 4] + [5, 6, 7],
+                    "groups": ["a"] * 4 + ["b"] * 3,
+                }
+            ),
```

Comment:
nit but I think you can just parametrise over `index` and `groups`? the `pl.DataFrame(...)` part is the same in each case, and kind of bloats up the test code

Reply:
Fixed